InteractionBase/src/handlers/GenericHandler.cpp

```diff
@@ -621,7 +652,10 @@ void GenericHandler::mousePressEvent(Visualization::Item *target, QGraphicsScene
 	}
 
 	if (event->modifiers() == Qt::NoModifier)
+	{
 		target->moveCursor(Visualization::Item::MoveOnPosition, event->pos().toPoint());
+		event->ignore();
```

Comment:
As you suspected, this is a really bad idea. You shouldn't just handle an event, and at the same time pretend that you didn't :)

Definitely remove this, and I'll comment a bit later what's the right thing to do.